cmd/frontend/internal/app/updatecheck/client.go

```diff
@@ -185,6 +188,67 @@ func getAndMarshalAggregatedUsageJSON(ctx context.Context) (_ json.RawMessage, _
 	return serializedCodeIntelUsage, serializedSearchUsage, nil
 }
 
+func getDependencyVersions(ctx context.Context, logFunc func(string, ...interface{})) (json.RawMessage, error) {
+	var (
+		err error
+		dv  dependencyVersions
+	)
+	// get redis cache server version
+	dv.RedisCacheVersion, err = getRedisVersion(redispool.Cache.Dial)
+	if err != nil {
+		logFunc("updatecheck.getDependencyVersions: unable to connect to redis cache instance", "error", err)
+	}
+
+	// get redis store server version
+	dv.RedisStoreVersion, err = getRedisVersion(redispool.Store.Dial)
+	if err != nil {
+		logFunc("updatecheck.getDependencyVersions: unable to get redis store version", "error", err)
+	}
+
+	// get postgres version
+	err = dbconn.Global.QueryRowContext(ctx, "SELECT version()").Scan(&dv.PostgresVersion)
+	if err != nil {
+		logFunc("updatecheck.getDependencyVersions: unable to get postgres version", "error", err)
+	}
+	return json.Marshal(dv)
+}
+
+func getRedisVersion(dialFunc func() (redis.Conn, error)) (string, error) {
+	conn, err := dialFunc()
+	if err != nil {
+		return "", err
+	}
+	buf, err := redis.Bytes(conn.Do("INFO"))
+	if err != nil {
+		return "", err
+	}
+
+	m, err := parseRedisInfo(buf)
+	return m["redis_version"], err
+
+}
+func parseRedisInfo(buf []byte) (map[string]string, error) {
+	var (
+		lines = bytes.Split(buf, []byte("\n"))
+		m     = make(map[string]string, len(lines))
+	)
+
+	for _, line := range lines {
+		line = bytes.TrimSpace(line)
+		if bytes.HasPrefix(line, []byte("#")) || len(line) == 0 {
+			continue
+		}
+
+		parts := bytes.Split(line, []byte(":"))
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("expected a key:value line, got '%s'", string(line))
```

Comment:
`%q` wraps quotes for you:

```suggestion
			return nil, fmt.Errorf("expected a key:value line, got %q", string(line))
```